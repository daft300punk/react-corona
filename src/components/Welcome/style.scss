.welcome {
    text-align: center;
    animation: fadein 2s;
    position: absolute;
    top: calc(50% - 40px);
    left: 50%;
    transform: translate(-50%, -50%);
    font-family: sans-serif;

    span {
      letter-spacing: 4px;
      text-align: left;
    }

    h1 {
      margin: 0;
      padding: 0;
      font-size: 84px;
      background: -webkit-linear-gradient(0, #0773ba, #fff200);
      -webkit-background-clip: text;
      -webkit-text-fill-color: transparent;
      letter-spacing: -2px;
    }

    @keyframes fadein {
        from {
            opacity:0;
        }
        to {
            opacity:1;
        }
    }

    @media only screen
    and (min-width: 924px)
    and (max-width: 1220px) {
        h1 {
            font-size: 64px;
        }
    }

    @media only screen
    and (max-width: 924px) {
        h1 {
            font-size: 54px;
            line-height: 54px;
        }
    }
}
