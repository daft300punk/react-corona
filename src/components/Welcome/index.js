import React from 'react';
import './style.scss';

/*
  A simple component to show Welcome Animation on first load.
*/
const Welcome = () => {
  return (
    <div className="welcome">
      <h1>Topcoder Stats</h1>
    </div>
  );
};

export default Welcome;
