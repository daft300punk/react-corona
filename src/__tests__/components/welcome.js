import React from 'react';
import renderer from 'react-test-renderer';

import Welcome from '../../components/Welcome';

describe('Welcome Component', () => {
  it('renders Welcome correctly', () => {
    const component = renderer.create(
      <Welcome />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});