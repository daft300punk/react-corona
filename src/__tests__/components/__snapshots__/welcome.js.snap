exports[`Welcome Component renders Welcome correctly 1`] = `
<div
  className="welcome">
  <h1>
    Topcoder Stats
  </h1>
</div>
`;
